### DISCUSSION TITLE
Error 403 on getting websocket from pods list api

### DISCUSSION THREAD

**Discussion by maps1995** [okd-project/okd#1610] [General]
**Describe the bug**

I have a user with cluster-admin role in my cluster, when I want to get websocket from pods list api in watch mode, I get error 403, but as you know cluster-admin user has a full access.
I've checked my permission using oc command:

```
oc auth can-i watch pods
yes
```

this is the API i'm working with:

```
/api/v1/namespaces/${ns}/pods?watch=true
```

I've tried to use this api without websocket and I could get the correct response without any error, my request was like that:

```
curl  -k -H "Authorization: Bearer $TOKEN" -H 'Accept: application/json' -X GET "https://$API_SERVER:6443/api/v1/namespaces/${ns}/pods?watch=true"
```

Also I wanna mention that my token in websocket request is correct, I get the websocket from other kubernetes apis such as pod logs with this cluster-admin without problem, but I have problem in this particular api.

In addition I've enabled debug logging mode in api server but there wasn't any helpful log for my request.

**Version**

4.11.0-0.okd-2022-08-20-022919
4.12.0-0.okd-2023-04-16-041331

**How reproducible**

**Log bundle**

**Comment by vrutkovs** [comment]
Kubernetes API is not using websockets, it uses [JSON/HTTP](https://kubernetes.io/docs/reference/using-api/api-concepts/). Pod logs, however, are served via websockets

**Reply by maps1995** [reply]
@vrutkovs Thank you for your answer.
But I've checked the okd console and I've seen okd is using the same api over websocket for getting pods status, could you tell me, how do you handle that on okd console, please?

**Reply by vrutkovs** [reply]
It uses websocket to communicate with the console server, not API server direcly afaik

**Reply by maps1995** [reply]
> It uses websocket to communicate with the console server, not API server direcly afaik

Thank you so much
